x-pack/plugins/infra/public/components/waffle/group_name.tsx

```diff
@@ -6,12 +6,14 @@
 import { EuiLink, EuiToolTip } from '@elastic/eui';
 import React from 'react';
 import styled from 'styled-components';
-import { InfraWaffleMapGroup } from '../../lib/lib';
+import { InfraPathType } from '../../../common/graphql/types';
+import { InfraWaffleMapGroup, InfraWaffleMapOptions } from '../../lib/lib';
 
 interface Props {
-  onDrilldown: () => void;
+  onDrilldown: (filter: string) => void;
   group: InfraWaffleMapGroup;
   isChild?: boolean;
+  options: InfraWaffleMapOptions;
```

Comment:
We could avoid having to add a dependency on the full set of options by just passing what we need (interface segregation principle), i.e. the path segment that this group is responsible for.

Even more, we could keep `onDrilldown` a function without arguments and just pass a specific drilldown function to each group name. That way the group implementation would not have to change if the drilldown action would change in the future.

Reply:
Great minds think alike! While I was working on this, I had a strong urge to refactor the way options are being passed around. This is on my radar for the next release.